File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/interceptor/WhiteAccessTokenInterceptor.java
package cc.coopersoft.accesstoken.interceptor;

import cc.coopersoft.accesstoken.api.AccessTokenHolder;
import okhttp3.Response;

import java.io.IOException;
import java.util.List;

public class WhiteAccessTokenInterceptor extends AccessTokenInterceptor {

  private final List<String> whiteList;

  public WhiteAccessTokenInterceptor(AccessTokenHolder tokenHolder, List<String> whiteList) {
    super(tokenHolder);
    this.whiteList = whiteList;
  }


  @Override
  public Response intercept(Chain chain) throws IOException {
    if (whiteList.stream().anyMatch(s -> chain.request().url().toString().matches(s))){
      return accessTokenRequest(chain);
    }
    return chain.proceed(chain.request());
  }
}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/interceptor/WhiteAccessTokenInterceptor.kt
package cc.coopersoft.accesstoken.interceptor

import cc.coopersoft.accesstoken.api.AccessTokenHolder
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class WhiteAccessTokenInterceptor(
    tokenHolder: AccessTokenHolder?,
    private val whiteList: List<String>
) : AccessTokenInterceptor(
    tokenHolder!!
) {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return if (whiteList.stream()
                .anyMatch { s: String? ->
                    s?.let { chain.request().url().toString().matches(s.toRegex()) } == true
                }
        ) {
            accessTokenRequest(chain)
        } else chain.proceed(chain.request())
    }
}